var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -159,8 +162,10 @@ def install(self, spec, prefix):
                     'JULIA_CPU_TARGET=generic'
                 ]
             else:
+                target_str = str(spec.target).replace('_', '-')
                 options += [
-                    'JULIA_CPU_TARGET={0}'.format(spec.target)
+                    'MARCH={0}'.format(target_str),
```

Comment:
The `MARCH` is not really needed since the spack compiler will inject the flags for the selected target. In theory, we should be able to just set `MARCH` and let that set `JULIA_CPU_TARGET` but I think some older versions had issues with that.

Reply:
I do seem to remember seeing some issues for older Julia versions with `JULIA_CPU_TARGET` not being set correctly from `MARCH`.

Would you prefer to remove the `MARCH` setting here? That's easy enough to do.

Reply:
Yes, go ahead and do that, so the `-march` settings are not getting set twice during the build.

Reply:
Done.